include/common/enum_configure.h

```diff
@@ -96,15 +96,25 @@ enum WasmEdge_Proposal {
 namespace WasmEdge {
 
 /// Host Module Registration C++ enumeration class.
-enum class HostRegistration : uint8_t { Wasi = 0, WasmEdge_Process, Max };
+enum class HostRegistration : uint8_t {
+  Wasi = 0,
+  WasmEdge_Process,
+#ifdef WASMEDGE_BUILD_WASI_CRYPTO
```

Comment:
The HostRegistration should be a static one. So you should not use `ifdef` here to remove it. Instead, just keep it.